### PR TITLE
Update bluebook-law-review.csl

### PR DESCRIPTION
**Pull request by jamesftierney**
## CSL Styles Pull Request Template
You're about to create a pull request to the CSL **styles** repository.  
If you haven't done so already, see <https://github.com/citation-style-language/styles/blob/master/CONTRIBUTING.md> for instructions on how to contribute, and <https://github.com/citation-style-language/styles/blob/master/STYLE_REQUIREMENTS.md> for the requirements a style must meet before acceptance.  
In addition, please fill out the pull request template below.

### Description
This revision corrects several formatting errors in prior implementations and extends coverage to additional source types. Specific changes include: italicizing case names in subsequent short-form citations regardless of whether a title-short is stored, but not full first-occurrence citations; fixing the statute-citation macro's erroneous comma-delimited outer group (which produced 42 U.S.C. , § 1983) by switching to a space delimiter; correcting the et-al-min threshold from 4 to 3 per Bluebook R 30.2.3; eliminating a double-space artifact between the author group and source text by setting the outer citation group delimiter to an empty string; adding explicit bill/legislation and manuscript branches in the subsequent-citation block to prevent impermissible supra forms for statutes and procedural rules; adding a dedicated report branch for Restatements and Model Codes (title in small caps, § number from the Report Number field, institutional publisher in parenthetical); adding a webpage/post-weblog source branch with URL and "last visited" access macro; adding status-variable support in the article-journal branch for forthcoming articles; removing the <substitute> element from name-macro to prevent double-title output for authorless items; suppressing author output for legal_case, legislation, and bill types; removing a spurious <label form="verb-short"> from name-short-macro; suppressing URL and "last visited" output for case and statute citations; adding a legislation/bill branch to the container macro; and splitting bill and legislation into separate source branches so that Zotero Bill items render in Statutes at Large session-law format (title, Pub. L. No. number, § section, volume Stat. page (year)) while legislation items continue to use the codified-statute macro. 

### Checklist
- [ ] Check that you've added a link to the style you used as a template in the `<info>` block at the beginning of the file with `rel="template"`.  
- [x] Check that you've added a link to the style guidelines with `rel="documentation"`.  
- [x] Check that you've added yourself as the `<author>` of the style or `<contributor>` for a style update. 
- [ ] Check that you've used the correct terms or labels instead of hardcoding into affixes (e.g., `<text variable="page" prefix="pp. "/>`).
- [ ] Check that you've not used `<text value="...` if not absolutely necessary.
- [x] Check that you've not changed line 1 of the style.

### CODE REVIEW

File: bluebook-law-review.csl [hunk 1]
```diff
@@ -130,11 +128,59 @@
               <text macro="container"/>
               <text variable="page-first"/>
             </group>
-            <text variable="locator"/>
+            <choose>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: revert this -- no need to test for a variable before calling it: an empty variable will just not render, same as your test

File: bluebook-law-review.csl [hunk 2]
```diff
@@ -130,11 +128,59 @@
...
+        <text macro="statute-citation"/>
+      </else-if>
+      <else-if type="bill">
+        <group delimiter=", ">
+          <text variable="title"/>
+          <group delimiter=" ">
+            <text value="Pub. L. No."/>
+            <text variable="number"/>
+          </group>
+          <group prefix="§ ">
```
Comment (adam3smith) [inline bluebook-law-review.csl]: No need for a group around a single element -- remove the group, move the prefix directly to section

File: bluebook-law-review.csl [hunk 3]
```diff
@@ -130,11 +128,59 @@
...
+              <text variable="page-first"/>
+              <text variable="locator"/>
+            </group>
+          </group>
+        </group>
+        <text macro="consec-periodical-issuance" prefix=" (" suffix=")"/>
+      </else-if>
+      <else-if type="book">
+        <text variable="title" text-case="title" font-variant="small-caps"/>
+        <text variable="locator" prefix=" "/>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: BB would never put a locator label (like col., ch. etc.) in there? Locators are most commonly pages, but can be all sort of other pinpoints.

File: bluebook-law-review.csl [hunk 4]
```diff
@@ -130,11 +128,59 @@
...
         </group>
       </if>
+      <else-if type="legislation" match="any">
+        <text macro="statute-citation"/>
+      </else-if>
+      <else-if type="bill">
+        <group delimiter=", ">
+          <text variable="title"/>
+          <group delimiter=" ">
+            <text value="Pub. L. No."/>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: Won't this always put the Public Law Number on bills? My understanding would be that more often that not bills are cited without it and it gets included only when the bill has passed (but I could be wrong here)

File: bluebook-law-review.csl [hunk 5]
```diff
@@ -144,59 +190,31 @@
...
-              <group delimiter=" ">
-                <text value="§"/>
-                <text variable="section"/>
-              </group>
-            </group>
-          </group>
-        </group>
-      </else-if>
-      <else-if type="article-newspaper article-magazine" match="any">
+      <else-if type="article-newspaper article-magazine thesis" match="any">
```
Comment (adam3smith) [inline bluebook-law-review.csl]: why are you adding thesis here? That seems like a very awkward fit with the periodical item types

File: bluebook-law-review.csl [hunk 6]
```diff
@@ -144,59 +190,31 @@
...
-            <text macro="container"/>
-          </group>
-          <text macro="issuance"/>
+          <text macro="container"/>
+          <text macro="periodical-issuance"/>
           <group delimiter=" ">
             <text value="at"/>
             <text variable="page-first"/>
           </group>
+          <choose>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: Same as above, this test isn't necessary.

File: bluebook-law-review.csl [hunk 7]
```diff
@@ -207,17 +225,36 @@
...
-        <text variable="locator" prefix=" "/>
-        <text macro="issuance" prefix=" (" suffix=")"/>
+      <else-if type="webpage post-weblog" match="any">
+        <group delimiter=", ">
+          <text variable="title" text-case="title" font-style="italic"/>
+          <text variable="container-title"/>
+          <text macro="periodical-issuance"/>
+        </group>
+      </else-if>
+      <else-if type="report">
```
Comment (adam3smith) [inline bluebook-law-review.csl:260]: What exactly are you envisioning this for? Reports in Zotero/CSL are typically reports by government non/government organizations, very much not legal reporters

File: bluebook-law-review.csl [hunk 8]
```diff
@@ -328,24 +398,44 @@
...
-                <group delimiter=" ">
-                  <text variable="volume"/>
-                  <text variable="container-title" form="short"/>
-                  <text macro="at_page"/>
+                <text macro="author-short"/>
+                <group>
+                  <text value="supra" font-style="italic" suffix=" "/>
+                  <text value="note" suffix=" "/>
+                  <text variable="first-reference-note-number"/>
+                  <group>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: Same as above, this entire group isn't needed.

File: bluebook-law-review.csl [hunk 9]
```diff
@@ -360,17 +450,38 @@
...
-                  <text value="note"/>
-                  <text variable="first-reference-note-number"/>
-                  <text macro="at_page"/>
+                  <choose>
+                    <if type="legal_case" match="any">
+                      <group>
+                        <text variable="volume" suffix=" "/>
+                        <text variable="container-title"/>
+                        <group>
+                          <text value="at" prefix=" " suffix=" "/>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: We prefer using terms where available, so 
`<text term="at".../>` 
with everything else the same.

File: bluebook-law-review.csl [hunk 10]
```diff
@@ -328,24 +398,44 @@
...
               <group delimiter=", ">
-                <text variable="title" form="short" font-style="italic"/>
-                <group delimiter=" ">
-                  <text variable="volume"/>
-                  <text variable="container-title" form="short"/>
-                  <text macro="at_page"/>
+                <text macro="author-short"/>
+                <group>
+                  <text value="supra" font-style="italic" suffix=" "/>
+                  <text value="note" suffix=" "/>
```
Comment (adam3smith) [inline bluebook-law-review.csl]: Also for note -- please use the term rather than the raw text value form.

**Review by adam3smith — CHANGES_REQUESTED** [review]
This is great! -- I have a couple of questions that might affect actual output and a couple of requests to improve/simplify the code.

For reference for others and posterity: https://papers.ssrn.com/sol3/papers.cfm?abstract_id=6416679

Since you already have reference examples in the library, would you consider putting them into a Zotero group? We have a group where we have APA and Chicago Manual examples, it'd make sense to have BB in there, too.

File: bluebook-law-review.csl (hunk 9 above)
Comment (jamesftierney) [reply bluebook-law-review.csl]: Thanks — I noted a couple other places where a term could be swapped in for a text value.

File: bluebook-law-review.csl (hunk 4 above)
Comment (jamesftierney) [reply bluebook-law-review.csl]: This is correct. As we only want Pub. L. No. for session laws, that can be printed conditional on the number variable being populated (i.e., enter the session law number only if you want to cite the session law). If the number variable is absent, the next update will use the codified citation instead.

File: bluebook-law-review.csl (hunk 3 above)
Comment (jamesftierney) [reply bluebook-law-review.csl]: Next version will implement this.

File: bluebook-law-review.csl (hunk 5 above)
Comment (jamesftierney) [reply bluebook-law-review.csl]: Your objection is correct, and the next version adds a new version that correctly implements thesis in Bluebook form:
> Author, Title (year) (Type, Institution)

File: bluebook-law-review.csl (hunk 7 above)
Comment (jamesftierney) [reply bluebook-law-review.csl:260]: This objection is also correct. The current report block was targeting C.F.R.-style regulatory citations — it uses § number prefix and (publisher year) with only a year. But that's wrong for most things report would actually hold in Zotero. While report is not a legal reporter, it is for government/institutional publications — working papers, agency reports, technical reports. That actually makes it a reasonable fit for Federal Register entries and similar administrative law materials, since those ARE government publications. There's no better CSL type for them. As the existing implementation was wrong for this use case, the next version keeps report but repurposes it specifically for a subset of administrative law materials since it's the best-fit CSL type. Any true C.F.R.-style citation that was previously using report should instead use legislation under the next version.

**Comment by jamesftierney** [comment]
Per reviewer's request, added a public Zotero group library that shows conventions for data-entry here: https://www.zotero.org/groups/6479791/bluebook_law_review_style_sample_citations

**Comment by jamesftierney** [comment]
Just checking on the status of this pull request in response to reviewer's requests for updates?

File: bluebook-law-review.csl
Comment (shelbaum) [inline bluebook-law-review.csl:115]: The currently active version of this style has \<choose\> tags around this block, making sure that the last visited date parenthetical only shows up if there's no other publication date provided, per Indigo Book R33.5 and Bluebook R18.2.2(c). Could that be restored to this version? As is I think this would incorrectly include (last visited) any time there is a URL.  
(PS, I just came across your paper documenting these updates.  Extremely useful, many thanks for improving the style!)
